libs/shared/src/containers/TransactionList/index.tsx

```diff
@@ -53,77 +77,75 @@ export const TransactionList = (props: MyProps) => {
-          </Table.Cell>
-          <Table.Cell textAlign='right' width={3}><div className={`font-big`}>{balanceCad} $&nbsp;&nbsp;</div></Table.Cell>
-        </Table.Row>
-    )});
 
+  function onDateRangeChange(from: Date, until: Date) {
+    setFromDate(from);
+    setUntilDate(until);
+  }
+  const { locale } = useSelector(selectLocale);
+
+  const maxRowCount = 50;
+  const { rates } = props;
+  const account = useActiveAccount();
+  const transactions = account!.history;
+  const transactionLoading = account?.historyLoading;
+
+  let filteredTx = transactions.filter((tx) => tx.date.toMillis() >= fromDate.getTime() && tx.date.toMillis() <= untilDate.getTime())
+  filteredTx.reverse();
+  let [ txOutput, jsxFooter ] = buildPagination(filteredTx, maxRowCount, 0);
+
+  let txJsxRows = txOutput.map((tx, index) => {
+    const change = fiatChange(tx, rates);
+    const balance = tx.balance *  weBuyAt(rates, tx.date.toJSDate());
+    const changeCad = toHuman(change, true);
+    const balanceCad = toHuman(balance, true);
+
+    const imgForLine = changeCad < 0 ? iconCard : iconThecoin;
+    const classForMoneyCell = changeCad < 0 ? styles.moneyNegative : styles.moneyPositive;
+    const contentForComment = changeCad < 0 ? <FormattedMessage {...sent} /> : <FormattedMessage {...received} />;
+    const descForComment = changeCad < 0 ? <FormattedMessage {...to} /> : <FormattedMessage {...from} />;
+
+    const monthTodisplay = tx.date.setLocale(locale).monthShort;
+    const yearToDisplay = tx.date.setLocale(locale).year;
+    const dayToDisplay = tx.date.setLocale(locale).day;
+
+    const timeToDisplay = tx.date.setLocale(locale).toLocaleString(DateTime.TIME_SIMPLE);
+    const addressComment = tx.counterPartyAddress;
     return (
-      <React.Fragment>
-        <DateRangeSelect onDateRangeChange={onDateRangeChange} />
-
-        <Dimmer.Dimmable>
-          <Dimmer active={transactionLoading}>Loading...</Dimmer>
-        </Dimmer.Dimmable>
-
-        <Table basic='very' singleLine >
-          <Table.Body>
-            {...txJsxRows}
-          </Table.Body>
-          {jsxFooter}
-        </Table>
-
-      </React.Fragment>
-    );
+      <TransactionLine
+        key={index}
+        locale={locale}
+
+        yearToDisplay={yearToDisplay}
+        monthTodisplay={monthTodisplay}
+        dayToDisplay={dayToDisplay}
+      
+        imgForLine={imgForLine}
+        contentForComment={contentForComment}
+        addressComment={addressComment}
+        descForComment={descForComment}
+      
+        classForMoneyCell={classForMoneyCell}
+        changeCad={changeCad}
+        timeToDisplay={timeToDisplay}
+      
+        balanceCad={balanceCad}
+      />
+  )});
+  const transactionsListZone = txJsxRows.length > 0 
+      ? <Grid padded>{...txJsxRows}{jsxFooter}</Grid> 
+      : <Segment placeholder><Header as="h4" icon><Icon name='search' /><FormattedMessage {...notransactions} /></Header></Segment> ;
```

Comment:
Added a "no transactions to display" zone it's always better to have that: users won't wonder if the thing is broken.

Reply:
Yes!  Very important, cause that's the users first impression!